web3/providers/auto.py

```diff
@@ -25,12 +25,12 @@ def load_provider_from_environment():
     return load_provider_from_uri(uri_string)
 
 
-def load_provider_from_uri(uri_string):
+def load_provider_from_uri(uri_string, headers=None):
```

Comment:
This seems to be fine for now but we *might* want to abstract this in the future to just be `provider_init_kwargs` and have it just be a pass-through, though as @carver has mentioned in the past, these types of passthroughs can end up being really hard to track and reason about, especially in any situation where there is more than a single level of it happening.